afl-fuzz.c

```diff
@@ -485,6 +490,26 @@ static void bind_to_free_cpu(void) {
   }
 
   closedir(d);
+#else
+  struct kinfo_proc *procs;
+  size_t nprocs;
+  size_t proccount;
+  cpuset_t c;
+  int s_name[3] = {CTL_KERN, KERN_PROC, KERN_PROC_ALL};
+  if (sysctl(s_name, 3, NULL, &nprocs, NULL, 0) < 0) return;
+  proccount = nprocs / sizeof(*procs);
+  procs = ck_alloc(nprocs);
+  if (sysctl(s_name, 3, NULL, &nprocs, NULL, 0) < 0) goto procs_free;
+
+  for (i = 0; i < proccount; i ++) {
+    if (procs[i].ki_oncpu < sizeof(cpu_used))
+      cpu_used[procs[i].ki_oncpu] = 1;
+  }
+
+procs_free:
```

Comment:
Can you write this without a `goto please? Just branch on the line 502

Reply:
sure